src/sources/datadog/agent/traces.rs

```diff
@@ -0,0 +1,214 @@
+use std::{collections::BTreeMap, sync::Arc};
+
+use bytes::Bytes;
+use chrono::{TimeZone, Utc};
+use futures::future;
+use http::StatusCode;
+use ordered_float::NotNan;
+use prost::Message;
+use warp::{filters::BoxedFilter, path, path::FullPath, reply::Response, Filter, Rejection, Reply};
+
+use crate::{
+    event::{Event, TraceEvent, Value},
+    internal_events::EventsReceived,
+    sources::{
+        datadog::agent::{self, handle_request, ApiKeyQueryParams, DatadogAgentSource},
+        util::ErrorMessage,
+    },
+    vector_core::ByteSizeOf,
+    SourceSender,
+};
+
+mod dd_proto {
+    include!(concat!(env!("OUT_DIR"), "/dd_trace.rs"));
+}
+
+pub(crate) fn build_warp_filter(
+    acknowledgements: bool,
+    multiple_outputs: bool,
+    out: SourceSender,
+    source: DatadogAgentSource,
+) -> BoxedFilter<(Response,)> {
+    build_trace_filter(acknowledgements, multiple_outputs, out, source)
+        .or(build_stats_filter())
+        .unify()
+        .boxed()
+}
+
+fn build_trace_filter(
+    acknowledgements: bool,
+    multiple_outputs: bool,
+    out: SourceSender,
+    source: DatadogAgentSource,
+) -> BoxedFilter<(Response,)> {
+    warp::post()
+        .and(path!("api" / "v0.2" / "traces" / ..))
```

Comment:
Do we also need to be handling the 0.3, 0.4, and 0.5 routes?

I noticed this when reviewing the opentelemetry collector PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5836/files#diff-6aa1e95c9a8e2db01693227b36333de7feea837fd59feb7c7c2727f4faf08f5aR64-R66

Reply:
Thoses routes are for tracing lib <-> agent trafic, we may support those at some point (note really sure this is relevant outside a large user specific usecase as opentelemtry tracing libs largely covers that). As of today (AFAIK) the trafic trace agent<->ddog is only using two urls, one for [traces](https://github.com/DataDog/datadog-agent/blob/a1413f3d9721b502ba12571c13866438f1ff8508/pkg/trace/writer/trace.go#L30) and the other for [apm stats](https://github.com/DataDog/datadog-agent/blob/a1413f3d9721b502ba12571c13866438f1ff8508/pkg/trace/writer/stats.go#L31).

Reply:
Aha 👍 Thanks for explaining.